Examples/clim-fig.lisp

```diff
@@ -128,55 +138,68 @@
-                                   (copy-from-pixmap canvas-pixmap 0 0 pixmap-width pixmap-height pane 0 0)
-                                   (draw-figure pane
-                                                x1 y1 x y
-                                                :fastp t
-						:cp-x1 cp-x1 :cp-y1 cp-y1
-						:cp-x2 cp-x2 :cp-y2 cp-y2)))
+                   (declare (ignore window))
+                   (set-status-line (format nil "~:(~A~) from (~D,~D) to (~D,~D)"
+                                            (slot-value *application-frame*
+                                                        'drawing-mode)
+                                            (round x1) (round y1)
+                                            (round x) (round y)))
+                   (with-output-recording-options (pane :record nil)
+                     (copy-from-pixmap canvas-pixmap 0 0
+                                       pixmap-width pixmap-height pane 0 0)
+                     (draw-figure pane
+                                  x1 y1 x y
+                                  :fastp t
+				  :cp-x1 cp-x1 :cp-y1 cp-y1
+				  :cp-x2 cp-x2 :cp-y2 cp-y2)))
 		(:pointer-button-release (&key event x y)
-                                         (when (= (pointer-event-button event)
-                                                  +pointer-left-button+)
-                                           (return-from processor (values x y))))
+                  (when (= (pointer-event-button event)
+                           +pointer-left-button+)
+                    (return-from processor (values x y))))
                 (:pointer-button-press (&key event x y)
-				       (cond
-					 ((= (pointer-event-button event)
-					     +pointer-right-button+)
-					   (setf cp-x1 x cp-y1 y))
-					 ((= (pointer-event-button event)
-					     +pointer-middle-button+)
-					   (setf cp-x2 x cp-y2 y)))))))
+		  (cond
+		    ((= (pointer-event-button event)
+			+pointer-right-button+)
+		     (setf cp-x1 x cp-y1 y))
+		    ((= (pointer-event-button event)
+			+pointer-middle-button+)
+		     (setf cp-x2 x cp-y2 y)))))))
       (set-status-line " ")
       (copy-from-pixmap canvas-pixmap 0 0 pixmap-width pixmap-height pane 0 0)
       (deallocate-pixmap canvas-pixmap)
-      (let ((new-record (with-output-to-output-record (*standard-output*)
-                          (draw-figure pane x1 y1 x y
-		                       :cp-x1 cp-x1 :cp-y1 cp-y1
-		                       :cp-x2 cp-x2 :cp-y2 cp-y2))))
-        (push new-record (clim-fig-undo-list *application-frame*))
-        (with-output-as-presentation (pane nil 'figure)
-            (stream-add-output-record *standard-output* new-record))
-        (replay new-record *standard-output* (bounding-rectangle new-record)))
-      (setf (clim-fig-redo-list *application-frame*) nil))))
+      (let ((new-presentation (with-output-as-presentation (pane nil 'figure)
```

Comment:
in com-redo you output the record wrapped in with-output-as-presentation, but here you push the record already being a presentation on the undo list. doesn't that mean, that during redo you output a presentation, whose object is another presentation, whose object is the actual figure?

Reply:
Yes, this was a leftover from 9d350e, thank you.